tests/e2e.rs

```diff
@@ -1,130 +1,73 @@
 
 #[tokio::test]
 async fn e2e_channel_creation() {
     let tc_client = testcontainers::clients::Cli::default();
     let bitcoind = Bitcoind::new(&tc_client, "0.19.1").unwrap();
-    bitcoind.init(5).await.unwrap();
-
-    let alice_wallet = Wallet::new("alice", bitcoind.node_url.clone())
-        .await
-        .unwrap();
-    let bob_wallet = Wallet::new("bob", bitcoind.node_url.clone()).await.unwrap();
-
-    {
-        // TODO: Could pass in the wallet and generate the address inside?
-        let address = alice_wallet.0.new_address().await.unwrap();
-        bitcoind
-            .mint(address, Amount::from_btc(3.0).unwrap())
-            .await
-            .unwrap();
-    }
-
-    {
-        let address = bob_wallet.0.new_address().await.unwrap();
-        bitcoind
-            .mint(address, Amount::from_btc(3.0).unwrap())
-            .await
-            .unwrap()
-    };
 
     let time_lock = 1;
-    let (channel_balance_alice, channel_balance_bob) = { (Amount::ONE_BTC, Amount::ONE_BTC) };
-
-    let alice0 = Alice0::new(channel_balance_alice, time_lock);
-    let bob0 = Bob0::new(channel_balance_bob, time_lock);
-
-    let message0_alice = alice0.next_message();
-    let message0_bob = bob0.next_message();
+    let (alice_balance, bob_balance) = (Amount::ONE_BTC, Amount::ONE_BTC);
 
-    let alice1 = alice0.receive(message0_bob, &alice_wallet).await.unwrap();
-    let bob1 = bob0.receive(message0_alice, &bob_wallet).await.unwrap();
+    let CreateActors {
+        alice,
+        alice_wallet,
+        bob,
+        bob_wallet,
+    } = make_create_actors(&bitcoind, alice_balance, bob_balance, time_lock).await;
 
-    let message1_alice = alice1.next_message();
-    let message1_bob = bob1.next_message();
+    let Created { alice, bob } = run_create_protocol(&alice_wallet, alice, &bob_wallet, bob).await;
 
-    let alice2 = alice1.receive(message1_bob).unwrap();
-    let bob2 = bob1.receive(message1_alice).unwrap();
+    assert_eq!(alice.signed_TX_f, bob.signed_TX_f);
 
-    let message2_alice = alice2.next_message();
-    let message2_bob = bob2.next_message();
-
-    let alice3 = alice2.receive(message2_bob).unwrap();
-    let bob3 = bob2.receive(message2_alice).unwrap();
-
-    let message3_alice = alice3.next_message();
-    let message3_bob = bob3.next_message();
-
-    let alice4 = alice3.receive(message3_bob).unwrap();
-    let bob4 = bob3.receive(message3_alice).unwrap();
+    alice_wallet
+        .0
+        .send_raw_transaction(alice.signed_TX_f.clone())
+        .await
+        .unwrap();
+}
 
-    let message4_alice = alice4.next_message();
-    let message4_bob = bob4.next_message();
+#[tokio::test]
+async fn e2e_channel_update() {
+    let tc_client = testcontainers::clients::Cli::default();
+    let bitcoind = Bitcoind::new(&tc_client, "0.19.1").unwrap();
 
-    let alice5 = alice4.receive(message4_bob).unwrap();
-    let bob5 = bob4.receive(message4_alice).unwrap();
+    let time_lock = 1;
```

Comment:
Are we able to make it work with the slightly higher timelock? I feel like if it works with `1` there is little guarantee it'd work with `2` :)

Reply:
We definitely need to properly test timelocks. @bonomat manually tested (successfully) some stuff yesterday, but we didn't merge anything yet.

I would argue that it should be a different test focusing on timelocks though.

Reply:
On that topic, just a reminder: use 
`tokio::time::delay_for(std::time::Duration::from_millis(30_000)).await;`
over 
`thread::sleep(std::time::Duration::from_millis(30_000));`
The latter one is blocking and should not be used in an async function.